### PR TITLE
Report an error when global variable does not exist

### LINKED ISSUE #1493: avg <non-existing-var> does not return an error
Filing so i remember:

`avg asdsadsadas` where `asdadsadsa` does not exist just returns nothing. It should give a user-visible error so I know something is wrong.

Reference: https://github.com/rizinorg/rizin/issues/333

### PR DESCRIPTION
**Pull request by Basstorm**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

Closes https://github.com/rizinorg/rizin/issues/1493

### CODE REVIEW

**Review by ret2libc — DISMISSED** [review]
Not really. The command itself should return RZ_CMD_STATUS_ERROR. With this patch it will print a message, but the command will be executed as if everything was fine.

**Review by ret2libc — COMMENTED** [review]
Could you do something similar for `avgt` and `avgn` as well?

File: librz/core/canalysis.c [hunk 1]
```diff
@@ -6868,19 +6868,20 @@ RZ_IPI char *rz_core_analysis_all_vars_display(RzCore *core, RzAnalysisFunction
...
 	RzAnalysisVarGlobal *glob = NULL;
 	if (name) {
 		global_vars = rz_list_new();
 		if (!global_vars) {
-			return;
+			return false;
 		}
 		glob = rz_analysis_var_global_get_byname(analysis, name);
 		if (!glob) {
+			RZ_LOG_ERROR("Global variable %s does not exist!\n", name);
```
Comment (wargio) [inline librz/core/canalysis.c]:
```suggestion
			RZ_LOG_ERROR("Global variable '%s' does not exist!\n", name);
```

File: librz/core/cmd_analysis.c [hunk 2]
```diff
@@ -8153,7 +8155,7 @@ RZ_IPI RzCmdStatus rz_analysis_global_variable_retype_handler(RzCore *core, int
 	const char *type = argv[2];
 	RzAnalysisVarGlobal *glob = rz_analysis_var_global_get_byname(core->analysis, name);
 	if (!glob) {
-		RZ_LOG_ERROR("No such global variable!\n");
+		RZ_LOG_ERROR("Global variable %s does not exist!\n", name);
```
Comment (wargio) [inline librz/core/cmd_analysis.c]:
```suggestion
		RZ_LOG_ERROR("Global variable '%s' does not exist!\n", name);
```

File: test/db/cmd/cmd_avg [hunk 3]
```diff
@@ -99,3 +99,15 @@ global char * foo @ 0x100
...
+
+NAME=avg <non-existing-var>
+FILE==
+CMDS=<<EOF
+avg foo
+EOF
+EXPECT=<<EOF
+EOF
+EXPECT_ERR=<<EOF
+ERROR: Global variable foo does not exist!
```
Comment (wargio) [inline test/db/cmd/cmd_avg]:
```suggestion
ERROR: Global variable 'foo' does not exist!
```

**Review by wargio — CHANGES_REQUESTED** [review]

**Review by wargio — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

**Comment by ret2libc** [comment]
Great!! Thanks a lot ;)

**Pull request merged**